pkg/manager/member/tikv_member_manager.go

```diff
@@ -559,28 +559,25 @@ func (tkmm *tikvMemberManager) setStoreLabelsForTiKV(tc *v1alpha1.TidbCluster) (
 	return setCount, nil
 }
 
-func (tkmm *tikvMemberManager) getNodeLabels(nodeName string) (map[string]string, error) {
+func (tkmm *tikvMemberManager) getNodeLabels(nodeName string, storeLabels []string) (map[string]string, error) {
 	node, err := tkmm.nodeLister.Get(nodeName)
 	if err != nil {
 		return nil, err
 	}
-	if ls := node.GetLabels(); ls != nil {
-		labels := map[string]string{}
-		if region, found := ls["region"]; found {
-			labels["region"] = region
-		}
-		if zone, found := ls["zone"]; found {
-			labels["zone"] = zone
-		}
-		if rack, found := ls["rack"]; found {
-			labels["rack"] = rack
+	labels := map[string]string{}
+	ls := node.GetLabels()
+	for _, storeLabel := range storeLabels {
+		if value, found := ls[storeLabel]; found {
+			labels[storeLabel] = value
+			continue
 		}
-		if host, found := ls[apis.LabelHostname]; found {
-			labels["host"] = host
+		if storeLabel == "host" {
```

Comment:
Add a TODO note that this needs to be changed when PD supports label key containing slash character.